### DISCUSSION TITLE
Brew Update fails with fetching homebrew-core and homebrew-bundle

### DISCUSSION THREAD

**Discussion by sivaswami** [Homebrew/discussions#298] [Everyday usage]
Machine: Linux (Ubuntu 20.04)
Brew was working earlier. All of a sudden it fails.

• • • brew update
fatal: unable to access 'https://github.com/Homebrew/brew/': error setting certificate verify locations:  CAfile: /home/linuxbrew/.linuxbrew/etc/openssl@1.1/cert.pem CApath: /home/linuxbrew/.linuxbrew/etc/openssl@1.1/certs
fatal: unable to access 'https://github.com/Homebrew/linuxbrew-core/': error setting certificate verify locations:  CAfile: /home/linuxbrew/.linuxbrew/etc/openssl@1.1/cert.pem CApath: /home/linuxbrew/.linuxbrew/etc/openssl@1.1/certs
fatal: unable to access 'https://github.com/Homebrew/homebrew-bundle/': error setting certificate verify locations:  CAfile: /home/linuxbrew/.linuxbrew/etc/openssl@1.1/cert.pem CApath: /home/linuxbrew/.linuxbrew/etc/openssl@1.1/certs
Error: Fetching $BREW_ROOT/BREW failed!
Fetching $BREW_ROOT/BREW/Library/Taps/homebrew/homebrew-core failed!
Fetching $BREW_ROOT/BREW/Library/Taps/homebrew/homebrew-bundle failed!

**Comment by gromgit** [comment]
What's the output of `brew config` and `brew doctor`?

Also...
```
Error: Fetching $BREW_ROOT/BREW failed!
```
Please don't obfuscate command output like this. It doesn't protect you in any way, and just serves to confuse everyone who's helping you.

**Reply by sivaswami** [reply]
Adding the output of brew config and brew doctor below.

• • • brew config
HOMEBREW_VERSION: 2.6.0-76-gcb38fe6
ORIGIN: https://github.com/Homebrew/brew
HEAD: cb38fe6e3ff18f2515661e4bba0e35a670f4888a
Last commit: 12 days ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core
Core tap HEAD: b725a0ba7ea4595cdbc5b0300e821b0550518863
Core tap last commit: 12 days ago
Core tap branch: master
HOMEBREW_PREFIX: /home/sivaswami/WORK/STOP/SOFT/BREW
HOMEBREW_REPOSITORY: /home/sivaswami/WORK/STOP/SOFT/BREW
HOMEBREW_CELLAR: /home/sivaswami/WORK/STOP/SOFT/BREW/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_EDITOR: code
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /home/sivaswami/WORK/STOP/SOFT/BREW/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 11.0 build (parse error)
Git: 2.29.2 => /home/sivaswami/WORK/STOP/SOFT/BREW/bin/git
Curl: 7.68.0 => /usr/bin/curl
Kernel: Linux 5.4.0-58-generic x86_64 GNU/Linux
OS: Ubuntu 20.04.1 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 9.3.0
/usr/bin/ruby: N/A
glibc: N/A
gcc: 5.5.0_7
xorg: N/A


• • • brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  isl@0.18

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/libpng16-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/icu-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/python3.6m-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/krb5-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/curl-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/xslt-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/python3.6-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/python3-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/pcre-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/libpng-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/xml2-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/freetype-config
  /home/sivaswami/WORK/STOP/SOFT/ANACONDA3/bin/ncursesw6-config

Warning: Your Homebrew's prefix is not /home/linuxbrew/.linuxbrew.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/home/linuxbrew/.linuxbrew).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.

**Reply by gromgit** [reply]
Your `HOMEBREW_PREFIX` is non-standard, but your installed `openssl@1.1` keg is using the standard prefix. Did you install Homebrew with the standard prefix, then move it over to its current location? If so, you've just broken a lot of your installed formulae.

The simplest hack to resolve this:
```
sudo ln -s /home/sivaswami/WORK/STOP/SOFT/BREW /home/linuxbrew/.linuxbrew
```
Homebrew will still continue to warn you about a non-standard prefix, but at least it'll get you going again.
